src/engine.vala

```diff
@@ -362,32 +363,32 @@ class KkcEngine : IBus.Engine {
     void apply_preferences () {
         Variant? variant;
 
-        variant = preferences.get ("punctuation_style");
+        variant = settings.get_value ("punctuation-style");
         assert (variant != null);
         context.punctuation_style = (Kkc.PunctuationStyle) variant.get_int32 ();
```

Comment:
I don't understand why you don't use binding here, while `context.punctuation_style` etc. are actually GObject properties:
https://ueno.github.io/libkkc/c/KkcContext.html#KkcContext.properties